addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -1,1107 +1,1031 @@
+        },
+    };
+
+    // hide top legend when too many items for device size
+    const MAX_LEGEND_LENGTH = 4 * (Math.max(1, config.device.size_class));
+
+    class GraphRenderer extends OwlAbstractRenderer {
+
+        constructor() {
+            super(...arguments);
+            this.canvasRef = useRef("canvas");
+            this._computeDerivedProps();
         }
-    },
-    /**
-     * This function creates a custom HTML tooltip.
-     *
-     * @private
-     * @param {Object} tooltipModel see chartjs documentation
-     */
-    _customTooltip: function (tooltipModel) {
-        this.$el.css({ cursor: 'default' });
-        if (this.$tooltip) {
-            this.$tooltip.remove();
+
+        mounted() {
+            if (!this.noContentHelperData){
+                this._renderChart();
+            }
         }
-        if (tooltipModel.opacity === 0) {
-            return;
+
+        patched() {
+            if (!this.noContentHelperData){
+                this._renderChart();
+            }
         }
-        if (tooltipModel.dataPoints.length === 0) {
-            return;
+
+        async willUpdateProps(nextProps) {
+            if (this.chart) {
+                this.chart.destroy(); // prevent Chart.js unecessary updates
+            }
+            this._computeDerivedProps(nextProps);
         }
 
-        if (this._isRedirectionEnabled()) {
-            this.$el.css({ cursor: 'pointer' });
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+
+        /**
+         * This function aims to remove a suitable number of lines from the tooltip in order to make
+         * it reasonably visible. A message indicating the number of lines is added if necessary.
+         * @private
+         * @param {Number} maxTooltipHeight this the max height in pixels of the tooltip
+         */
+        _adjustTooltipHeight(maxTooltipHeight) {
+            const sizeOneLine = this.tooltip.querySelector('tbody tr').clientHeight;
+            const tbodySize = this.tooltip.querySelector('tbody').clientHeight;
+            let toKeep = Math.floor((maxTooltipHeight - (this.tooltip.clientHeight - tbodySize)) / sizeOneLine) - 1;
+            toKeep = Math.max(0, toKeep); // used to avoid toKeep = -1
+            const lines = this.tooltip.querySelectorAll('tbody tr');
+            const toRemove = lines.length - toKeep;
+            if (toRemove > 0) {
+                for (let i = toKeep; i < lines.length; ++i) {
+                    lines[i].remove();
+                }
+                const tr = document.createElement('tr');
+                const td = document.createElement('td');
+                tr.classList.add('o_show_more');
+                td.innerHTML = _t("...");
+                tr.appendChild(td);
+                this.tooltip.querySelector('tbody').appendChild(tr);
+            }
```

Comment:
This is the kind of stuff we don't like to do. Can't we trim the content of the tooltip before feeding it? And define proper css rules to handle the case where the trimmed content overflows, s.t. we have a scrollbar or something?

Reply:
I could maybe fix the height of the tooltip elements in order to be able to compute how many items should be kept before
rendering the tooltip.

Reply:
But we should then keep scss and js aligned

Reply:
I don't like that idea

Reply:
(we don't want a scrollbar)

Reply:
Yep I know... I don't like the bricolage we do here :(

Reply:
After discussion, we have agreed to keep actual code